src/asm/section.cpp

```diff
@@ -508,6 +508,14 @@ void sect_AlignPC(uint8_t alignment, uint16_t offset)
 					- offset) % alignSize) {
 		error("Section's alignment fails required alignment (offset from section start = $%04"
 			PRIx32 ")\n", curOffset);
+	} else if (alignment >= 16) {
+		// Treat an alignment large enough as fixing the address.
+		// Note that this also ensures that a section's alignment never becomes 16 or greater.
+		if (alignment > 16) {
+			error("Alignment must be between 0 and 16, not %u\n", alignment);
```

Comment:
`getSection` caps the alignment at 16 when it does this error; so probably this should too. Then a test case `align 17, 42` would error (so fail to assemble) but continue non-fatally like `align 16, 42`.
```suggestion
			error("Alignment must be between 0 and 16, not %u\n", alignment);
			alignment = 16;
```

Reply:
That would be a dead write, since `alignment` is not used afterwards.

Reply:
Okay, I see now; it already treats alignment >16 that way.